guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/list/RepositoryItemPresenter.java

```diff
@@ -71,6 +71,11 @@ public void setRepository( final Repository repository,
         view.refresh();
     }
 
+    public void refreshBranches() {
+        populateBranches( view.getSelectedBranch() );
+        view.refresh();
```

Comment:
I would prefer keep this invocation for this case, since the Select component is sometimes tricky.